src/provisioner/provisioner.py

```diff
@@ -255,11 +255,15 @@ def _provision_components(cortx_conf_url: str, _conf_idx: str, interfaces: Enum,
 
     @staticmethod
     def _apply_consul_config(_conf_idx: str):
-        num_endpoints = Conf.get(_conf_idx, 'cortx>external>consul>num_endpoints')
-        for idx in range(0, num_endpoints):
-            if 'http' in Conf.get(_conf_idx, f'cortx>external>consul>endpoints[{idx}]'):
-                consul_endpoint = Conf.get(_conf_idx, f'cortx>external>consul>endpoints[{idx}]')
-                break
+        try:
+            num_endpoints = int(Conf.get(_conf_idx, 'cortx>external>consul>num_endpoints'))
+            for idx in range(0, num_endpoints):
+                if 'http' in Conf.get(_conf_idx, f'cortx>external>consul>endpoints[{idx}]'):
+                    consul_endpoint = Conf.get(_conf_idx, f'cortx>external>consul>endpoints[{idx}]')
+                    break
```

Comment:
What if consul_endpoint is not found ? Isnt it an error ? 
```
for ...
  if ...
    consul_endpoint = ..
if consul_endpoint == None:
   throw ProvisionerError(....)
```

Reply:
in if statement we are checking string 'http' in endpoint so if it is not found it will not be consider.

Reply:
Handle the scenario in case consul_endpoint is empty/None.